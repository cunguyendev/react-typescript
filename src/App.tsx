import React from 'react';
import Header from './layouts/header';
import Footer from './layouts/footer';
import Wrapper from './layouts/wrapper';

const theApp = {
  name:  'Code Lab',
  baseUrl: '/'
}

const navigationData = [
  {
    url: '/',
    text: 'Home',
    active: true
  },
  {
    url: '/about/',
    text: 'About'
  },
  {
    url: '/projects/',
    text: 'Projects'
  }
]

const socialData = [
  {
    host: 'https://www.facebook.com/',
    username: 'facebookexample',
    text: 'Facebook',
  },
  {
    host: 'https://www.twitter.com/',
    username: 'twitterexample',
    text: 'Twitter',
  },
]

const App = () => {
  return (
    <div className="App">
      <Header navigationData={navigationData} theApp={theApp}/>
      <Wrapper />
      <Footer socialData={socialData} theApp={theApp}/>
    </div>
  );
}

export default App;
