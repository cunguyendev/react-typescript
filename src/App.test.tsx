import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Header from './layouts/header';
import Wrapper from './layouts/wrapper';
import 'jest-styled-components';

describe('Loading component', () => {
  test('Test snapshot', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render component correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Wrapper)).toHaveLength(1);
  });
});
