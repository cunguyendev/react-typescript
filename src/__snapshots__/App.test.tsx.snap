// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Test snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "action": [Function],
      "authenticated": false,
    }
  }
>
  <styled.div>
    <Memo(Header)
      theApp={
        Object {
          "baseUrl": "/",
          "name": "Code Lab",
        }
      }
    />
    <Memo(Wrapper) />
  </styled.div>
</ContextProvider>
`;
